ddtrace/appsec/_iast/_taint_tracking/aspects.py

```diff
@@ -1242,3 +1235,99 @@ def re_expand_aspect(orig_function: Optional[Callable], flag_added_args: int, *a
         iast_propagation_error_log(f"re_expand_aspect. {e}")
 
     return result
+
+
+def ospathjoin_aspect(*args: Any, **kwargs: Any) -> Any:
+    if not all(isinstance(arg, IAST.TEXT_TYPES) for arg in args):
+        return os.path.join(*args, **kwargs)
+
+    try:
+        return _aspect_ospathjoin(*args, **kwargs)
+    except Exception as e:
+        iast_propagation_error_log(f"ospathjoin_aspect. {e}")
+
+    return os.path.join(*args, **kwargs)
```

Comment:
I wonder if we'll reduce the performance impact with
```suggestion
def ospathjoin_aspect(*args: Any, **kwargs: Any) -> Any:
    if all(isinstance(arg, IAST.TEXT_TYPES) for arg in args):
      try:
          return _aspect_ospathjoin(*args, **kwargs)
      except Exception as e:
          iast_propagation_error_log(f"ospathjoin_aspect. {e}")

    return os.path.join(*args, **kwargs)
```